com.unity.render-pipelines.universal/Runtime/DeferredLights.cs

```diff
@@ -435,6 +436,8 @@ internal bool AccurateGbufferNormals
         ProfilingSampler m_ProfilingSamplerDeferredFogPass = new ProfilingSampler(k_DeferredFogPass);
         ProfilingSampler m_ProfilingSamplerClearStencilPartialPass = new ProfilingSampler(k_ClearStencilPartial);
 
+        // TODO: move? share with forward? Pass in?
+        private LightCookieManager m_LightCookieManager;
```

Comment:
I think it should be shared somewhere.
Does each instance of LightCookeManager creates its own texture atlas?

Reply:
Yes, they do. I'll move it somewhere common and pass it in as deferred the integration didn't require any specialization.

Reply:
Hmm, we would need to add new public api to pass the CookieManager in and if it isn't passed then the cookies would fail.
Luckily, the LightCookieManager does a lazy init, so the atlas isn't allocated until you have at least 1 cookie.
But it's true that if you switch or somehow use both paths (forward, deferred), then there's 2 copies of the atlas and shader data.

Reply:
In deferred rendering mode, ForwardLight code will still run for forward-only and transparent passes, so presumably the 2 instances of LightCookieManager may have allocated their atlas?

Reply:
Ok. It's now moved and shared.